react/features/device-selection/functions.js

```diff
@@ -29,3 +42,143 @@ export function getDeviceSelectionDialogProps(stateful: Object | Function) {
         selectedVideoInputId: settings.cameraDeviceId
     };
 }
+
+/**
+ * Processes device requests from external applications.
+ *
+ * @param {Dispatch} dispatch - The redux {@code dispatch} function.
+ * @param {Function} getState - The redux function that gets/retrieves the redux
+ * state.
+ * @param {Object} request - The request to be processed.
+ * @param {Function} responseCallback - The callback that will send the
+ * response.
+ * @returns {boolean}
+ */
+export function processRequest( // eslint-disable-line max-params
+        dispatch: Dispatch<any>,
+        getState: Function,
+        request: Object,
+        responseCallback: Function) {
+    if (request.type === 'devices') {
```

Comment:
If the processing only happens when request.type is devices, it makes sense to me to return early when it's not to avoid the extra nesting, especially in a longer block like this one where it can be easy to forget when the if block has exited.